testing/sample_framework/src/android/android_app_framework.cc

```diff
@@ -478,6 +479,44 @@ std::string ReadTextInput(const char* title, const char* message,
   return g_text_entry_field_data->ReadText(title, message, placeholder);
 }
 
+void SetEnvironmentVariableFromStringExtra(JNIEnv* env, const char* extra_name,
+                                           jobject intent) {
+  jclass intent_class = env->GetObjectClass(intent);
+  jmethodID get_string_extra = env->GetMethodID(
+      intent_class, "getStringExtra", "(Ljava/lang/String;)Ljava/lang/String;");
+  env->DeleteLocalRef(intent_class);
+
+  jstring extra_name_jstring = env->NewStringUTF(extra_name);
+  jstring extra_value_jstring = (jstring)env->CallObjectMethod(
+      intent, get_string_extra, extra_name_jstring);
+  env->DeleteLocalRef(extra_name_jstring);
+
+  if (extra_value_jstring != nullptr) {
+    const char* extra_value =
+        env->GetStringUTFChars(extra_value_jstring, nullptr);
+    setenv(extra_name, extra_value == nullptr ? "" : extra_value,
+           /*overwrite=*/1);
+    env->ReleaseStringUTFChars(extra_value_jstring, extra_value);
```

Comment:
Since `extra_value` is potentially `nullptr`, I think you should conditionally invoke `ReleaseStringUTFChars()` only if it is not null. Alternately, just assume that `GetStringUTFChars()` returns non-null and get rid of the `extra_value == nullptr` check above. It would be some incredibly rare situation where it would return null and it would indicate a much larger problem anyways.

Reply:
Done.